Digital Credentials: implement base OpenID4VP IDL structure

marcoscaceres opened the pull request:
#### 46a449a46e3ee95220bcfd35ede0dff2634eade8
<pre>
Digital Credentials: implement base OpenID4VP IDL structure
<a href="https://bugs.webkit.org/show_bug.cgi?id=277670">https://bugs.webkit.org/show_bug.cgi?id=277670</a>
<a href="https://rdar.apple.com/problem/133265654">rdar://problem/133265654</a>

Reviewed by Chris Dumez and Abrar Rahman Protyasha.

Adds basic OpenID4VPRequest IDL structure to use by Digital Credentials API.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/identity/IdentityRequestProvider.h:
* Source/WebCore/Modules/identity/IdentityRequestProvider.idl:
* Source/WebCore/Modules/identity/OpenID4VPRequest.h: Copied from Source/WebCore/Modules/identity/IdentityRequestProvider.idl.
* Source/WebCore/Modules/identity/OpenID4VPRequest.idl: Copied from Source/WebCore/Modules/identity/IdentityRequestProvider.idl.
* Source/WebCore/Sources.txt:

Canonical link: <a href="https://commits.webkit.org/282036@main">https://commits.webkit.org/282036@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b4d3b9307e7827ba933e0288265b0cf0e1600ac7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/61820 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/41174 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/14412 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/65799 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/12365 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/63939 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/48860 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/12636 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/49826 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/8565 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/64889 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/38238 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/53544 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/30658 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/34884 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/10776 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/11296 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/56693 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/11079 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/67528 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/5763 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/10818 "Found 1 new test failure: workers/worker-to-worker.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/57206 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/5788 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/53493 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/57451 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13755 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/4727 "Passed tests") | | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/36974 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/38058 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/39154 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/37803 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/Modules/identity/IdentityRequestProvider.h:
```diff
@@ -26,14 +26,16 @@
 
 namespace WebCore {
 
+using DigitalCredentialRequest = std::variant<OpenID4VPRequest>;
```
Why are we using a variant if there is a single possible type?

aprotyas replied:
As you guessed in a later comment, this will need to be a `std::variant` eventually, but I agree that we don't need it right now.

aprotyas replied:
Also, I think instead of including `OpenID4VPRequest.h`, you can get away with a forward declaration for `OpenID4VPRequest`, if you do keep the `std::variant` in this PR.

marcoscaceres replied:
Will drop the variant aspect until such time as another variant gets standardized and just set it to OpenID4VPRequest for now.

cdumez commented on Source/WebCore/Modules/identity/IdentityRequestProvider.idl:
```diff
@@ -23,7 +23,9 @@
  * THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+typedef (OpenID4VPRequest) DigitalCredentialRequest;
```
Why the parentheses?

marcoscaceres replied:
this was for `(X or Y)`... which I'll get rid of.

cdumez reviewed (approved):
Anyway, LGTM but I don't think we need a variant at the moment.

marcoscaceres committed:
Digital Credentials: implement base OpenID4VP IDL structure
https://bugs.webkit.org/show_bug.cgi?id=277670
rdar://problem/133265654

Reviewed by Chris Dumez and Abrar Rahman Protyasha.

Adds basic OpenID4VPRequest IDL structure to use by Digital Credentials API.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/identity/IdentityRequestProvider.h:
* Source/WebCore/Modules/identity/IdentityRequestProvider.idl:
* Source/WebCore/Modules/identity/OpenID4VPRequest.h: Copied from Source/WebCore/Modules/identity/IdentityRequestProvider.idl.
* Source/WebCore/Modules/identity/OpenID4VPRequest.idl: Copied from Source/WebCore/Modules/identity/IdentityRequestProvider.idl.
* Source/WebCore/Sources.txt:

Canonical link: https://commits.webkit.org/282036@main
